pkg/probe/grpc/grpc.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package grpc
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"google.golang.org/grpc"
+	grpchealth "google.golang.org/grpc/health/grpc_health_v1"
+	"google.golang.org/grpc/metadata"
+	"k8s.io/component-base/version"
+	"k8s.io/kubernetes/pkg/probe"
+)
+
+var (
+	errGrpcNotServing = errors.New("GRPC_STATUS_NOT_SERVING")
+)
+
+// Prober is an interface that defines the Probe function for doing GRPC readiness/liveness checks.
+type Prober interface {
+	Probe(host, service string, port int, timeout time.Duration, opts ...grpc.DialOption) (probe.Result, string, error)
+}
+
+type grpcProber struct {
+}
+
+// New Prober for execute grpc probe
+func New() Prober {
+	return grpcProber{}
+}
+
+// Probe executes a grpc call to check the liveness/readiness of container.
+// Returns the Result status, command output, and errors if any.
+func (p grpcProber) Probe(host, service string, port int, timeout time.Duration, opts ...grpc.DialOption) (probe.Result, string, error) {
+	v := version.Get()
+
+	md := metadata.New(map[string]string{
+		"User-Agent": fmt.Sprintf("kubernetes/%s.%s", v.Major, v.Minor),
```

Comment:
I know it was defined like this in KEP. What do you think making it consistent with httpGet's `kube-probe/1.22` instead of `kubernetes/1.22`?

CC @bowei

Reply:
Unified define is fine,WDYT @bowei